### PR TITLE
MULE-10814: Unify coordinate language for ArtifactIsolatedClassLoader…

### PR DESCRIPTION
**Pull request by gsfernandes**
…Builder

### CODE REVIEW

File: tests/runner/src/main/java/org/mule/test/runner/classification/AbstractPatternDependencyFilter.java [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+public class AbstractPatternDependencyFilter implements DependencyFilter {
+
+  private final Set<String> patterns = new HashSet<String>();
+
+  /**
+   * Creates the filter using the Maven coordinates
+   *
+   * @param coords that define the inclusion patterns
+   */
+  public AbstractPatternDependencyFilter(final Collection<String> coords) {
```
Comment (pablokraan) [inline tests/runner/src/main/java/org/mule/test/runner/classification/AbstractPatternDependencyFilter.java]: coords > coordinates

File: tests/runner/src/main/java/org/mule/test/runner/classification/AbstractPatternDependencyFilter.java [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+    this(asList(coords));
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public boolean accept(final DependencyNode node, List<DependencyNode> parents) {
+    final Dependency dependency = node.getDependency();
+    if (dependency == null) {
```
Comment (pablokraan) [inline tests/runner/src/main/java/org/mule/test/runner/classification/AbstractPatternDependencyFilter.java]: Why a null dependency is accepted?

File: tests/runner/src/main/java/org/mule/test/runner/classification/AbstractPatternDependencyFilter.java [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+      return true;
+    }
+    return accept(dependency.getArtifact());
+  }
+
+  protected boolean accept(final Artifact artifact) {
```
Comment (pablokraan) [inline tests/runner/src/main/java/org/mule/test/runner/classification/AbstractPatternDependencyFilter.java]: Add javadoc or make method private

File: tests/runner/src/main/java/org/mule/test/runner/classification/AbstractPatternDependencyFilter.java [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+    }
+    return false;
+  }
+
+  private boolean accept(final Artifact artifact, final String pattern) {
+    final String[] tokens =
+        new String[] {artifact.getGroupId(), artifact.getArtifactId(), artifact.getExtension(), artifact.getClassifier(),
+            artifact.getBaseVersion()};
+
+    final String[] patternTokens = pattern.split(":");
```
Comment (pablokraan) [inline tests/runner/src/main/java/org/mule/test/runner/classification/AbstractPatternDependencyFilter.java]: Define constant for ":"

File: tests/runner/src/main/java/org/mule/test/runner/classification/AbstractPatternDependencyFilter.java [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+
+    // fail immediately if pattern tokens outnumber tokens to match
```
Comment (pablokraan) [inline tests/runner/src/main/java/org/mule/test/runner/classification/AbstractPatternDependencyFilter.java]: Start comments with upper case

File: tests/runner/src/main/java/org/mule/test/runner/classification/AbstractPatternDependencyFilter.java [hunk 6]
```diff
@@ -0,0 +1,154 @@
...
+    }
+
+    return matched;
+  }
+
+  private boolean matches(final String token, final String pattern) {
+    boolean matches;
+
+    // support full wildcard and implied wildcard
+    if ("*".equals(pattern) || pattern.length() == 0) {
```
Comment (pablokraan) [inline tests/runner/src/main/java/org/mule/test/runner/classification/AbstractPatternDependencyFilter.java]: Add constant for "*"

Comment (pablokraan) [inline tests/runner/src/main/java/org/mule/test/runner/classification/AbstractPatternDependencyFilter.java]: Would be more readable if you define functions for each condition and remove the comments. For example: supportsFullAndImpliedWilcard(pattern)

File: tests/runner/src/main/java/org/mule/test/runner/classification/AbstractPatternDependencyFilter.java [hunk 7]
```diff
@@ -0,0 +1,154 @@
...
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public int hashCode() {
+    int hash = 17;
+    hash = hash * 31 + patterns.hashCode();
+    return hash;
```
Comment (pablokraan) [inline tests/runner/src/main/java/org/mule/test/runner/classification/AbstractPatternDependencyFilter.java]: Why not to return just patterns.hashcode?

File: tests/runner/src/main/java/org/mule/test/runner/classification/PatternExclusionsDependencyFilter.java [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+ * @since 4.0
+ */
+public class PatternExclusionsDependencyFilter extends AbstractPatternDependencyFilter {
+
+  /**
+   * Creates the filter using the Maven coordinates
+   *
+   * @param coords that define the inclusion patterns
+   */
+  public PatternExclusionsDependencyFilter(final Collection<String> coords) {
```
Comment (pablokraan) [inline tests/runner/src/main/java/org/mule/test/runner/classification/PatternExclusionsDependencyFilter.java]: coords > coordinates

File: tests/runner/src/main/java/org/mule/test/runner/classification/PatternExclusionsDependencyFilter.java [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+ * </pre>
+ *
+ * @since 4.0
+ */
+public class PatternExclusionsDependencyFilter extends AbstractPatternDependencyFilter {
+
+  /**
+   * Creates the filter using the Maven coordinates
+   *
+   * @param coords that define the inclusion patterns
```
Comment (pablokraan) [inline tests/runner/src/main/java/org/mule/test/runner/classification/PatternExclusionsDependencyFilter.java]: inclusion > exclusion?

File: tests/runner/src/main/java/org/mule/test/runner/classification/PatternExclusionsDependencyFilter.java [hunk 10]
```diff
@@ -0,0 +1,56 @@
...
+   */
+  public PatternExclusionsDependencyFilter(final Collection<String> coords) {
+    super(coords);
+  }
+
+  /**
+   * Creates the filter using the Maven coordinates
+   *
+   * @param coords that define the inclusion patterns
```
Comment (pablokraan) [inline tests/runner/src/main/java/org/mule/test/runner/classification/PatternExclusionsDependencyFilter.java]: inclusion > exclusion?

File: tests/runner/src/main/java/org/mule/test/runner/classification/AbstractPatternDependencyFilter.java [hunk 11]
```diff
@@ -0,0 +1,154 @@
...
+ * Base class for pattern filters.
+ * <p/>
+ * The artifact pattern syntax is of the form:
+ * <pre>
+ * [groupId]:[artifactId]:[extension]:[classifier]:[version]
+ * </pre>
+ *
+ * @since 4.0
+ */
+public class AbstractPatternDependencyFilter implements DependencyFilter {
```
Comment (pablokraan) [inline tests/runner/src/main/java/org/mule/test/runner/classification/AbstractPatternDependencyFilter.java]: Add unit tests for this class and subclasses

**Comment by gsfernandes** [comment]
@pablokraan, changes were made. PR is ready for review.

**Pull request merged**